<HTML>
<HEAD>
<meta charset="UTF-8">
<title>FlowException - corda</title>
<link rel="stylesheet" href="../../../style.css">
</HEAD>
<BODY>
<a href="../../index.html">corda</a>&nbsp;/&nbsp;<a href="../index.html">net.corda.core.flows</a>&nbsp;/&nbsp;<a href="./index.html">FlowException</a><br/>
<br/>
<h1>FlowException</h1>
<code><span class="keyword">open</span> <span class="keyword">class </span><span class="identifier">FlowException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="../../net.corda.core/-corda-exception/index.html"><span class="identifier">CordaException</span></a></code>
<p>Exception which can be thrown by a <a href="../-flow-logic/index.html">FlowLogic</a> at any point in its logic to unexpectedly bring it to a permanent end.
The exception will propagate to all counterparty flows and will be thrown on their end the next time they wait on a
<a href="../-flow-session/receive.html">FlowSession.receive</a> or <a href="../-flow-session/send-and-receive.html">FlowSession.sendAndReceive</a>. Any flow which no longer needs to do a receive, or has already
ended, will not receive the exception (if this is required then have them wait for a confirmation message).</p>
<p>If the <em>rethrown</em> <a href="./index.html">FlowException</a> is uncaught in counterparty flows and propagation triggers then the exception is
downgraded to an <a href="../-unexpected-flow-end-exception/index.html">UnexpectedFlowEndException</a>. This means only immediate counterparty flows will receive information
about what the exception was.</p>
<p><a href="./index.html">FlowException</a> (or a subclass) can be a valid expected response from a flow, particularly ones which act as a service.
It is recommended a <a href="../-flow-logic/index.html">FlowLogic</a> document the <a href="./index.html">FlowException</a> types it can throw.</p>
<h3>Constructors</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="-init-.html">&lt;init&gt;</a></p>
</td>
<td>
<code><span class="identifier">FlowException</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.flows.FlowException$<init>(kotlin.String, kotlin.Throwable)/message">message</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span><span class="symbol">, </span><span class="identifier" id="net.corda.core.flows.FlowException$<init>(kotlin.String, kotlin.Throwable)/cause">cause</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span><span class="symbol">)</span></code><br/>
<code><span class="identifier">FlowException</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.flows.FlowException$<init>(kotlin.String)/message">message</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span><span class="symbol">)</span></code><br/>
<code><span class="identifier">FlowException</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.flows.FlowException$<init>(kotlin.Throwable)/cause">cause</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span><span class="symbol">)</span></code><br/>
<code><span class="identifier">FlowException</span><span class="symbol">(</span><span class="symbol">)</span></code><code><span class="identifier">FlowException</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.flows.FlowException$<init>(kotlin.String, kotlin.Throwable, kotlin.Long)/message">message</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span><span class="symbol">, </span><span class="identifier" id="net.corda.core.flows.FlowException$<init>(kotlin.String, kotlin.Throwable, kotlin.Long)/cause">cause</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span><span class="symbol">, </span><span class="identifier" id="net.corda.core.flows.FlowException$<init>(kotlin.String, kotlin.Throwable, kotlin.Long)/originalErrorId">originalErrorId</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-long/index.html"><span class="identifier">Long</span></a><span class="symbol">?</span>&nbsp;<span class="symbol">=</span>&nbsp;null<span class="symbol">)</span></code>
<p>Exception which can be thrown by a <a href="../-flow-logic/index.html">FlowLogic</a> at any point in its logic to unexpectedly bring it to a permanent end.
The exception will propagate to all counterparty flows and will be thrown on their end the next time they wait on a
<a href="../-flow-session/receive.html">FlowSession.receive</a> or <a href="../-flow-session/send-and-receive.html">FlowSession.sendAndReceive</a>. Any flow which no longer needs to do a receive, or has already
ended, will not receive the exception (if this is required then have them wait for a confirmation message).</p>
</td>
</tr>
</tbody>
</table>
<h3>Properties</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="original-error-id.html">originalErrorId</a></p>
</td>
<td>
<code><span class="keyword">var </span><span class="identifier">originalErrorId</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-long/index.html"><span class="identifier">Long</span></a><span class="symbol">?</span></code>
<p>the ID backing <a href="get-error-id.html">getErrorId</a>. If null it will be set dynamically by the flow framework when
    the exception is handled. This ID is propagated to counterparty flows, even when the <a href="./index.html">FlowException</a> is
    downgraded to an <a href="../-unexpected-flow-end-exception/index.html">UnexpectedFlowEndException</a>. This is so the error conditions may be correlated later on.</p>
</td>
</tr>
</tbody>
</table>
<h3>Inherited Properties</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-exception/cause.html">cause</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">val </span><span class="identifier">cause</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-exception/message.html">message</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">val </span><span class="identifier">message</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-exception/original-exception-class-name.html">originalExceptionClassName</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">var </span><span class="identifier">originalExceptionClassName</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-exception/original-message.html">originalMessage</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">val </span><span class="identifier">originalMessage</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span></code></td>
</tr>
</tbody>
</table>
<h3>Functions</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="get-error-id.html">getErrorId</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">getErrorId</span><span class="symbol">(</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-long/index.html"><span class="identifier">Long</span></a><span class="symbol">?</span></code></td>
</tr>
</tbody>
</table>
<h3>Inherited Functions</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-exception/add-suppressed.html">addSuppressed</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">addSuppressed</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.CordaException$addSuppressed(kotlin.Array((kotlin.Throwable)))/suppressed">suppressed</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-array/index.html"><span class="identifier">Array</span></a><span class="symbol">&lt;</span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">&gt;</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-unit/index.html"><span class="identifier">Unit</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-exception/equals.html">equals</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">equals</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.CordaException$equals(kotlin.Any)/other">other</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-any/index.html"><span class="identifier">Any</span></a><span class="symbol">?</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-boolean/index.html"><span class="identifier">Boolean</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-exception/hash-code.html">hashCode</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">hashCode</span><span class="symbol">(</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-int/index.html"><span class="identifier">Int</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-exception/set-cause.html">setCause</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">setCause</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.CordaException$setCause(kotlin.Throwable)/cause">cause</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-throwable/index.html"><span class="identifier">Throwable</span></a><span class="symbol">?</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-unit/index.html"><span class="identifier">Unit</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core/-corda-exception/set-message.html">setMessage</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">fun </span><span class="identifier">setMessage</span><span class="symbol">(</span><span class="identifier" id="net.corda.core.CordaException$setMessage(kotlin.String)/message">message</span><span class="symbol">:</span>&nbsp;<a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-string/index.html"><span class="identifier">String</span></a><span class="symbol">?</span><span class="symbol">)</span><span class="symbol">: </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-unit/index.html"><span class="identifier">Unit</span></a></code></td>
</tr>
</tbody>
</table>
<h3>Extension Functions</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="../../net.corda.core.utilities/kotlin.-any/context-logger.html">contextLogger</a></p>
</td>
<td>
<code><span class="keyword">fun </span><a href="https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-any/index.html"><span class="identifier">Any</span></a><span class="symbol">.</span><span class="identifier">contextLogger</span><span class="symbol">(</span><span class="symbol">)</span><span class="symbol">: </span><span class="identifier">&lt;ERROR CLASS&gt;</span></code>
<p>When called from a companion object, returns the logger for the enclosing class.</p>
</td>
</tr>
</tbody>
</table>
<h3>Inheritors</h3>
<table>
<tbody>
<tr>
<td>
<p><a href="../../net.corda.finance.flows/-two-party-trade-flow/-asset-mismatch-exception/index.html">AssetMismatchException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">AssetMismatchException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core.contracts/-attachment-resolution-exception/index.html">AttachmentResolutionException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">AttachmentResolutionException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code>
<p>The node asked a remote peer for the attachment identified by <a href="../../net.corda.core.contracts/-attachment-resolution-exception/hash.html">hash</a> because it is a dependency of a transaction
being resolved, but the remote peer would not provide it.</p>
</td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.testing.dsl/-attachment-resolution-exception/index.html">AttachmentResolutionException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">AttachmentResolutionException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core.contracts/-broken-attachment-exception/index.html">BrokenAttachmentException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">BrokenAttachmentException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code>
<p>A non-specific exception for the attachment identified by <a href="../../net.corda.core.contracts/-broken-attachment-exception/attachment-id.html">attachmentId</a>. The context
for this error is provided via the <a href="#">message</a> and <a href="#">cause</a>.</p>
</td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.finance.flows/-cash-exception/index.html">CashException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">CashException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.testing.dsl/-double-spent-inputs/index.html">DoubleSpentInputs</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">DoubleSpentInputs</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.testing.dsl/-duplicate-output-label/index.html">DuplicateOutputLabel</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">DuplicateOutputLabel</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core.contracts/-insufficient-balance-exception/index.html">InsufficientBalanceException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">InsufficientBalanceException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code>
<p>Thrown if a request is made to spend an amount of a <a href="../../net.corda.core.contracts/-fungible-asset/index.html">FungibleAsset</a> but there aren't enough tokens in the vault.</p>
</td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core.transactions/-missing-contract-attachments/index.html">MissingContractAttachments</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">MissingContractAttachments</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code>
<p>A contract attachment was missing when trying to automatically attach all known contract attachments</p>
</td>
</tr>
<tr>
<td>
<p><a href="../-notary-exception/index.html">NotaryException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">NotaryException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code>
<p>Exception thrown by the notary service if any issues are encountered while trying to commit a transaction. The
underlying <a href="../-notary-exception/error.html">error</a> specifies the cause of failure.</p>
</td>
</tr>
<tr>
<td>
<p><a href="../-state-replacement-exception/index.html">StateReplacementException</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">class </span><span class="identifier">StateReplacementException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core.node.services/-states-not-available-exception/index.html">StatesNotAvailableException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">StatesNotAvailableException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core.contracts/-transaction-resolution-exception/index.html">TransactionResolutionException</a></p>
</td>
<td>
<code><span class="keyword">open</span> <span class="keyword">class </span><span class="identifier">TransactionResolutionException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code>
<p>The node asked a remote peer for the transaction identified by <a href="../../net.corda.core.contracts/-transaction-resolution-exception/hash.html">hash</a> because it is a dependency of a transaction
being resolved, but the remote peer would not provide it.</p>
</td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core.contracts/-transaction-verification-exception/index.html">TransactionVerificationException</a></p>
</td>
<td>
<code><span class="keyword">abstract</span> <span class="keyword">class </span><span class="identifier">TransactionVerificationException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code>
<p>Indicates that some aspect of the transaction named by <a href="../../net.corda.core.contracts/-transaction-verification-exception/tx-id.html">txId</a> violates the platform rules. The exact type of failure
is expressed using a subclass. TransactionVerificationException is a <a href="./index.html">FlowException</a> and thus when thrown inside
a flow, the details of the failure will be serialised, propagated to the peer and rethrown.</p>
</td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.finance.flows/-two-party-trade-flow/-unacceptable-price-exception/index.html">UnacceptablePriceException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">UnacceptablePriceException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.finance.workflows.utils/-unknown-calendar/index.html">UnknownCalendar</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">UnknownCalendar</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code></td>
</tr>
<tr>
<td>
<p><a href="../../net.corda.core.node.services/-vault-query-exception/index.html">VaultQueryException</a></p>
</td>
<td>
<code><span class="keyword">class </span><span class="identifier">VaultQueryException</span>&nbsp;<span class="symbol">:</span>&nbsp;<a href="./index.html"><span class="identifier">FlowException</span></a></code></td>
</tr>
</tbody>
</table>
</BODY>
</HTML>
